File: algorithm/boj/_1000/1546.py
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    user_input = list(map(int, sys.stdin.readline().split()))
    li = []
    val_sum = 0
    
    for i in range(n):
        li.append(user_input[i])
    
    li.sort()
    
    for i in range(n):
        val_sum += round(li[i]/li[-1]*100, 6)
    
    print(round(val_sum/n, 6))
    

# 백준에서는 런타임에러 나는 코드
# if __name__ == "__main__":
#     n = int(sys.stdin.readline())
#     li = list(map(int, sys.stdin.readline().split()))[:4]
#     val_sum = 0
    
#     li.sort()
#     print(li)
    
#     for i in range(n):
#         val_sum += round(li[i]/li[-1]*100, 6)
    
#     print(round(val_sum/n, 6))

File: algorithm/boj/_1000/1546.java
package TIL.algorithm.boj._1000;

import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

class Main1546 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine());
        double valSum = 0;
        StringTokenizer st = new StringTokenizer(br.readLine());
        double[] arr = new double[n];
        
        for(int i=0; i<n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(arr);
        
        for(int i=0; i<n; i++) {
            valSum += arr[i]/arr[n-1]*100;
        }
        bw.write(String.format("%.6f", valSum/n)+"\n");
        br.close();
        bw.flush();
        bw.close();
    }
}